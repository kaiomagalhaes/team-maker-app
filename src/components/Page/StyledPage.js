import styled, { injectGlobal } from 'styled-components';
import theme from 'theme';

/* eslint-disable */
injectGlobal`
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    font-size: 1.5rem;
    line-height: 2;
    margin: 0;
    padding: 0;
  }
  a {
    color: ${theme.black};
    text-decoration: none;
  }
`;
/* eslint-enable */

export const Inner = styled.div`
  margin: 0 auto;
  max-width: ${props => props.theme.maxWidth};
  padding: 2rem;
`;

export default styled.div`
  background: ${props => props.theme.white};
  color: ${props => props.theme.black};
`;
