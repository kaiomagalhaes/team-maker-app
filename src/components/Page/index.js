import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import Header from 'components/Header';
import Meta from 'components/Meta';
import theme from 'theme';

import StyledPage, { Inner } from './StyledPage';

const propTypes = {
  children: PropTypes.element,
};

const defaultProps = {
  children: undefined,
};

const Page = ({ children }) => (
  <ThemeProvider theme={theme}>
    <StyledPage>
      <Meta />
      <Header />
      <Inner>{children}</Inner>
    </StyledPage>
  </ThemeProvider>
);

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default Page;
