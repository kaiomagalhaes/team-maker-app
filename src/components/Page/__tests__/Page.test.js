import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Page from '../index';

describe('Page', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<Page />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
