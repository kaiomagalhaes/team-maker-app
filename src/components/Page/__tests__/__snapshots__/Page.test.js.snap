// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders and matches the snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "black": "#000",
      "maxWidth": "1000px",
      "white": "#fff",
    }
  }
>
  <StyledPage>
    <Meta />
    <Header />
    <StyledPage__Inner />
  </StyledPage>
</ThemeProvider>
`;
